build/bazel/remote/execution/v2/remote_execution.proto

```diff
@@ -1288,6 +1315,218 @@ message WaitExecutionRequest {
+      // contents with the search pattern applied must be added to the proposective strong fingerprint.
+      // See comments on
+      // [ObservedPathSet][build.bazel.remote.execution.v2.ObservedPathSet] for how to generate a
+      // directory hash.
+      DIRECTORY_ENUMERATION = 1;
+
+      // `Path` is a directory that was enumerated by the action. If the directory is missing,
+      // this pathset does not match the filesystem. Otherwise the hash of the string constant "E"
+      // followed by a hash of the directory contents filtered to only the file names in
+      // ObservedAccessFileNames (compared to only the directory's filenames without extensions)
+      // must be added to the prospective strong fingerprint. See comments on
+      // [ObservedPathSet][build.bazel.remote.execution.v2.ObservedPathSet] for how to generate a
+      // directory hash.
+      DIRECTORY_SEARCH = 2;
+
+      // 'Path' is a file probed for existence, but not read, by the action and that was present
+      // during execution. If the file is not present or is instead a directory, this pathset
+      // does not match the filesystem. Otherwise the hash of the string constant "F" must be
+      // added to the prospective strong fingerprint.
+      EXISTING_FILE_PROBE = 3;
+
+      // `Path` is a directory probed for existence, but not enumerated, by the action and that
+      // was present during execution. If the directory is not present, or is instead a file,
+      // this pathset does not match the filesystem. Otherwise the hash of the string constant "D"
+      // must be added to the prospective strong fingerprint.
+      EXISTING_DIRECTORY_PROBE = 4;
+
+      // `Path` is a file or directory probed for existence by the action that was not
+      // present during execution. If path is present as a file or directory, this pathset
+      // does not match the filesystem. Otherwise the hash of the string constant "P" must
+      // be added to the prospective strong fingerprint.
+      ABSENT_FILE_OR_DIRECTORY_PROBE = 5;
+    }
+
+    // File or directory path.
+    string Path = 1;
+
+    // Defines the meaning of `Path` and how to compare to the local filesystem.
+    OpKind Kind = 2;
+
+    // For entries of `OpType.DIRECTORY_ENUMERATION`, the enumeration pattern used.
+    // When null/empty/missing the search pattern was "*" meaning the full directory.
+    string DirectorySearchPattern = 3;
+  }
+
+  // Observed path set entries. In order to ensure consistent hashing of the same `ObservedPathSet`,
+  // the entries MUST be sorted lexicographically by Path code point (or, equivalently, by UTF-8
+  // bytes).
+  repeated Entry Entries = 1;
+
+  // For directory comparisons in Entries marked with `OpKind.DIRECTORY_SEARCH`, defines the filenames,
+  // with extensions removed, that were read by the action. These are used as the filter for generating
+  // a directory hash to add to the prospective strong fingerprint.
```

Comment:
Clarification why no extensions: Various tools try multiple versions of a path (.h, .hpp, ...)

Reply:
Also .dll and .exe for assembly search. Headers: Also no-extension and other possibilities. We leave off the extensions because in actual implementation sandboxing can see the files actually probed or read. 

Also: Avoid oscillations of directory contents by avoiding extensions.

---
In reply to: [453824291](https://github.com/erikma/remote-apis/pull/1#discussion_r453824291) [](ancestors = 453824291)